proto/job_submission.proto

```diff
@@ -0,0 +1,179 @@
+  string submissionid = 3 [(google.api.field_behavior) = REQUIRED];
+}
+
+message GetJobLogRequest {
+  // Required. The namespace of the cluster for the job
+  string namespace = 1 [(google.api.field_behavior) = REQUIRED];
+  // Required. The name of the cluster for the job
+  string clustername = 2 [(google.api.field_behavior) = REQUIRED];
+  // Required. The submission id of the job
+  string submissionid = 3 [(google.api.field_behavior) = REQUIRED];
+}
+
+message GetJobLogReply {
+  // Content of the log. Always from the beginning
+  string log = 1;
+}
+
+message ListJobDetailsRequest {
+  // Required. The namespace of the cluster for the job
+  string namespace = 1 [(google.api.field_behavior) = REQUIRED];
+  // Required. The name of the cluster for the job
+  string clustername = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+message ListJobSubmissionInfo {
+  repeated JobSubmissionInfo submissions = 1;
+}
+
+message StopRayJobSubmissionRequest {
+  // Required. The namespace of the cluster for the job
+  string namespace = 1 [(google.api.field_behavior) = REQUIRED];
+  // Required. The name of the cluster for the job
+  string clustername = 2 [(google.api.field_behavior) = REQUIRED];
+  // Required. The submission id of the job
+  string submissionid = 3 [(google.api.field_behavior) = REQUIRED];
+}
+
+message DeleteRayJobSubmissionRequest {
+  // Required. The namespace of the cluster for the job
+  string namespace = 1 [(google.api.field_behavior) = REQUIRED];
+  // Required. The name of the cluster for the job
+  string clustername = 2 [(google.api.field_behavior) = REQUIRED];
+  // Required. The submission id of the job
+  string submissionid = 3 [(google.api.field_behavior) = REQUIRED];
+}
+
+
+// RayJobSubmission definition
+message RayJobSubmission {
+  // Required. Entry point
+  string entrypoint = 1 [(google.api.field_behavior) = REQUIRED];
+  // Optional submission id
+  string submission_id = 2;
+  // Optional. Arbitrary user-provided metadata for the job.
+  map<string, string> metadata = 3;
+  // Optional. The runtime environment for the job.  - yaml string.
+  string runtime_env = 4;
+  // Optional. Number of CPUs to allocate for the execution of the entrypoint command, separately from any Ray tasks or actors that are created by it.
+  float entrypoint_num_cpus = 5;
+  // Optional. Number of GPUs to allocate for the execution of the entrypoint command, separately from any Ray tasks or actors that are created by it.
+  float entrypoint_num_gpus = 6;
+  // Optional. The quantity of various custom resources to allocate for the execution of the entrypoint command, separately from any Ray tasks or actors that are created by it.
+  map<string, string> entrypoint_resources = 7;
+}
+
+message JobSubmissionInfo{
+  // Submission entry point
+  string entrypoint = 1;
+  // Job ID
+  string job_id = 2;
+  // Submission ID
+  string submission_id = 3;
+  // Submission status
+  string status = 4;
```

Comment:
`status` is required.

Reply:
We defined as required only the data that user provides data, doing it for data coming from the server does not make sense to me